File: coffee-system/src/main/java/com/jesperancinha/coffee/system/concurrency/PostActionCallableImpl.java
package com.jesperancinha.coffee.system.concurrency;

import com.jesperancinha.coffee.api.concurrency.PostActionCallable;
import com.jesperancinha.coffee.api.manager.MachineProcessor;
import com.jesperancinha.coffee.system.input.Employees.Employee.Actions.PostAction;
import com.jesperancinha.coffee.system.objects.ActionDescriptor;
import lombok.Getter;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.text.MessageFormat;
import java.util.concurrent.TimeUnit;

/**
 * Created by joaofilipesabinoesperancinha on 05-05-16.
 */
@Service
@Getter
@Slf4j
public class PostActionCallableImpl extends ActionCallable implements PostActionCallable {

    @Autowired
    private MachineProcessor machineProcessor;

    public PostActionCallableImpl(String name) {
        super(name);

    }

    @Override
    public void addPostAction(PostAction postAction) {
        this.actionDescriptorList
                .add(ActionDescriptor.builder().description(postAction.getDescription()).time(postAction.getTime())
                        .build());
    }

    public Boolean call() {
        this.actionDescriptorList.forEach(
                actionDescriptor -> {
                    log.info(MessageFormat.format("Ending with {0}", actionDescriptor.getDescription()));
                    try {
                        TimeUnit.MILLISECONDS.sleep(actionDescriptor.getTime());
                    } catch (InterruptedException e) {
                        log.error(e.getMessage(), e);
                    }
                }
        );
        return true;
    }

}

File: coffee-system/src/main/kotlin/org/jesperancinha/coffee/system/concurrency/PostActionCallableImpl.kt
package org.jesperancinha.coffee.system.concurrency

import lombok.Getter
import lombok.extern.slf4j.Slf4j
import org.jesperancinha.coffee.system.api.concurrency.QueueCallable
import org.jesperancinha.coffee.system.input.Employees.Employee.Actions.PostAction
import org.jesperancinha.coffee.system.manager.MachineProcessorImpl
import org.jesperancinha.coffee.system.objects.ActionDescriptor
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.text.MessageFormat
import java.util.concurrent.TimeUnit
import java.util.function.Consumer

/**
 * Created by joaofilipesabinoesperancinha on 05-05-16.
 */
@Service
@Getter
@Slf4j
class PostActionCallableImpl(name: String) : ActionCallable(name), QueueCallable {
    @Autowired
    private val machineProcessor: MachineProcessorImpl? = null
    fun addPostAction(postAction: PostAction) {
        actionDescriptorList
            .add(
                ActionDescriptor(
                    description = postAction.description,
                    time = postAction.time
                )
            )
    }

    override fun call(): Boolean {
        actionDescriptorList.forEach(
            Consumer { actionDescriptor: ActionDescriptor ->
                logger.info(MessageFormat.format("Ending with {0}", actionDescriptor.description))
                try {
                    TimeUnit.MILLISECONDS.sleep(actionDescriptor.time.toLong())
                } catch (e: InterruptedException) {
                    logger.error(e.message, e)
                }
            }
        )
        return true
    }

    companion object {
        val logger: Logger = LoggerFactory.getLogger(PostActionCallableImpl::class.java)
    }
}